packages/extension-api/src/extension-api.d.ts

```diff
@@ -170,6 +170,82 @@ declare module '@podman-desktop/api' {
     readonly secrets: SecretStorage;
   }
 
+  /**
+   * Represents an extension.
+   *
+   * To get an instance of an `Extension` use {@link extensions.getExtension getExtension}.
+   */
+  export interface Extension<T> {
+    /**
+     * The canonical extension identifier in the form of: `publisher.name`.
+     */
+    readonly id: string;
+
+    /**
+     * The uri of the directory containing the extension.
+     */
+    readonly extensionUri: Uri;
+
+    /**
+     * The absolute file path of the directory containing this extension. Shorthand
+     * notation for {@link Extension.extensionUri Extension.extensionUri.fsPath} (independent of the uri scheme).
+     */
+    readonly extensionPath: string;
+
+    /**
+     * The parsed contents of the extension's package.json.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    readonly packageJSON: any;
```

Comment:
could we expose the version as a property of Extension ? I would rather do `extension.version` with a nice typing than `extension.packageJSON.version?`

Reply:
I think it would be a separate enhancement not related to this one

Also I don't think an extension should check versions of other extensions

Reply:
I think they totally should ? Let's say AI Lab version 1.1 expose "GetInferenceServers()" but not version 1.0... I might want to handle that case, and preferably not by doing (extension.GetInferenceServers === undefined)

I don't really know how we want to manage typing, by I imagine the ai lab should publish @types package ? providing the methods types it is exporting ?